ClaudiaRojasSoto/QR_properties_app
# Initialize Rails Application for QR Properties App

## Overview
This Pull Request introduces the initial setup of the `qr_properties_app`, laying the foundational structure for our new real estate QR code generation application. The setup includes the basic Rails application structure, Gemfile configuration, database setup, and the initial README documentation.

## Changes
- Generated new Rails application using `rails new qr_properties_app` command.
- Configured `Gemfile` with essential gems for the project's initial stage, including `pg` for PostgreSQL integration, `rqrcode` for QR code generation, and other development tools like `rubocop` for code analysis.
- Setup `database.yml` for PostgreSQL database integration, ensuring compatibility with our planned production deployment environment.
- Added initial README with project description, setup instructions, and contribution guidelines to foster clear and structured development practices from the outset.

## Purpose
The purpose of setting up this initial Rails application structure is to provide a strong, configurable, and scalable foundation for developing the QR properties application. This setup enables us to start building out specific features like property management, QR code generation, and integration with external services like Google Drive for document storage.

## How to Test
1. Ensure PostgreSQL is running on your local machine.
2. Clone the repository and switch to the `feature/initial-setup` branch.
3. Run `bundle install` to install the required Ruby gems.
4. Setup the database by running `rails db:create db:migrate`.
5. Start the Rails server with `rails server` and visit `http://localhost:3000` to see the Rails welcome page.

## Next Steps
- [ ] Add the spec folders for testing
- [ ] Begin development of the property model and associated controllers.
- [ ] Integrate QR code generation functionality.
- [ ] Setup Active Storage with Google Drive integration for document and image storage.

Please review the changes and provide any feedback or approval. Let's ensure our project's foundation is solid as we proceed with building out the functionalities.

Thank you!
